plugins/performance-lab/includes/admin/plugin-activate-ajax.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Handles activation of Performance Features (Plugins) using AJAX.
+ */
+
+/* global perflabPluginActivateAjaxData */
+( function () {
+	// @ts-ignore
+	const { i18n, a11y } = wp;
+	const { __ } = i18n;
+
+	/**
+	 * Handles click events on elements with the class 'perflab-install-active-plugin'.
+	 *
+	 * This asynchronous function listens for click events on the document and executes
+	 * the provided callback function if triggered.
+	 *
+	 * @param {MouseEvent} event - The click event object that is triggered when the user clicks on the document.
+	 *
+	 * @return {Promise<void>} - The asynchronous function returns a promise that resolves to void.
+	 */
+	async function handlePluginActivationClick( event ) {
+		const target = /** @type {HTMLElement} */ ( event.target );
+
+		if ( ! target.classList.contains( 'perflab-install-active-plugin' ) ) {
+			return;
+		}
+
+		// Prevent the default link behavior.
+		event.preventDefault();
+
+		target.classList.add( 'updating-message' );
+		target.textContent = __( 'Activating…', 'performance-lab' );
+		target.style.pointerEvents = 'none';
+		target.parentElement.style.cursor = 'not-allowed';
+
+		a11y.speak( __( 'Activating…', 'performance-lab' ) );
+
+		const pluginSlug = target.getAttribute( 'data-plugin-slug' ).trim();
+
+		// Send an AJAX POST request to activate the plugin.
+		try {
+			const response = await fetch(
+				// @ts-ignore
+				perflabPluginActivateAjaxData.ajaxUrl,
```

Comment:
Why not use the REST API? You could just use `wp.apiFetch` then without having to export `perflabPluginActivateAjaxData` to the frontend.